File: fixtures/scale/scala/medium/src/calc/funcs/InverseTrig.scala
package calc.funcs

import calc.config.Settings
import calc.eval.FunctionRegistry

object InverseTrig {
  def register(registry: FunctionRegistry, settings: Settings): Unit = {
    registry.addUnary("asin", x => settings.fromRadians(math.asin(x)))
    registry.addUnary("acos", x => settings.fromRadians(math.acos(x)))
    registry.addUnary("atan", x => settings.fromRadians(math.atan(x)))
  }
}

File: fixtures/scale/kotlin/medium/src/calc/funcs/InverseTrig.kt
package calc.funcs

import calc.config.Settings
import calc.eval.FunctionRegistry
import kotlin.math.acos
import kotlin.math.asin
import kotlin.math.atan

fun registerInverseTrig(registry: FunctionRegistry, settings: Settings) {
    registry.addUnary("asin") { settings.fromRadians(asin(it)) }
    registry.addUnary("acos") { settings.fromRadians(acos(it)) }
    registry.addUnary("atan") { settings.fromRadians(atan(it)) }
}
